### PR TITLE
ys_col_note

### PR DESCRIPTION
**Pull request by kylebaron**
# Summary 
- Select some columns from a `yspec` object and create a column definition string
- Optionally append the unit; QUESTION - should this default to `TRUE`?
- Optionally convert to title case; QUESTION - should this default to `TRUE`?
- Optionally convert to a single string; this currently defaults to `TRUE`

@KatherineKayMRG I simplified this quite a bit based on initial prototyping. I felt like it wasn't that obvious what columns were going to come back in the note and that it was reasonable to just ask to name the columns to appear. This would (I think) also allow user to enforce an order which might be nice. 

# Example

``` r
library(yspec)
spec <- ys_help$spec()

ys_col_note(spec, ALT, AST, SCR, AMT, AAG, BLQ,  .to_string = TRUE)
#> [1] "ALT: alanine aminotransferase; AST: aspartate aminotransferase; SCR: serum creatinine; AMT: dose amount; AAG: alpha-1-acid glycoprotein; BLQ: below limit of quantification"
```

<sup>Created on 2022-03-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

File: R/col-note.R [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    sh <- unlist(ys_get_label(.spec))  
+  }
+  if(isTRUE(.title_case)) {
+    sh <- toTitleCase(sh)  
+  }
+  if(isTRUE(.unit)) {
+    u <- ys_get_unit(.spec, parens = TRUE)
+    sh <- paste0(sh, " ", u)
+  }
+  ans <- paste0(names(.spec), ": ", unname(sh))
```
[the remark this answers is not in the document]
Comment (kylebaron) [reply R/col-note.R]: Thanks @KatherineKayMRG ; we'll use `.sep` to separate column name and definition and `.collapse` to be the separator when we collapse it down.

**Comment by kylebaron** [comment]
@KatherineKayMRG Thanks for reviewing; I believe everything is up to date based on your review.

**Review by KatherineKayMRG — APPROVED** [review]
Thanks @kylebaron - Looks great!

**Comment by kylebaron** [comment]
Awesome!

**Pull request merged**